tokio-util/src/either.rs

```diff
@@ -164,6 +164,31 @@ where
     }
 }
 
+impl<L, R, Item, Error> futures_sink::Sink<Item> for Either<L, R>
+where
+    L: futures_sink::Sink<Item, Error = Error>,
+    R: futures_sink::Sink<Item, Error = Error>,
+{
+    type Error = Error;
+
+    fn poll_ready(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<()>> {
```

Comment:
In each method, you need this:
```suggestion
    fn poll_ready(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Error>> {
```
Right now, you're using `std::io::Error` instead, which is the wrong error type.

Reply:
How do I fix it ?